lib/path.js

```diff
@@ -746,13 +760,14 @@ const win32 = {
   },
 
   /**
-   * @param {string} path
+   * @param {PathLike} path
    * @param {string} [suffix]
    * @returns {string}
    */
   basename(path, suffix) {
     if (suffix !== undefined)
       validateString(suffix, 'ext');
+    path = toPathIfFileURL(path, { windows: true });
```

Comment:
Did we run any benchmarks to see the impact of this?

Reply:
*(resolving, as benchmark was executed)*